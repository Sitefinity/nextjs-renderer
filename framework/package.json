{
  "name": "sitefinity-react-renderer-framework",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@types/react": "18.2.28",
    "react": "18.2.0",
    "typescript": "5.2.2"
  },
  "scripts": {
    "build": "tsc"
  }
}
