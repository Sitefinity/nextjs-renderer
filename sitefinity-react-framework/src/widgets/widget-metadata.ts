import { EditorMetadata } from './widget-editor-metadata';

export interface WidgetMetadata {
    componentType: any;
    designerMetadata: any;
    selectorCategory?: string;
    editorMetadata?: EditorMetadata
}
