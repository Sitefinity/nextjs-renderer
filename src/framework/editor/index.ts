export * from './renderer-contract-interfaces';
export * from './services';
export * from './utils';
export * from './widgets';

