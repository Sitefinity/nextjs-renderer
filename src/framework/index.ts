export * from './editor';
export * from './rest-sdk';
