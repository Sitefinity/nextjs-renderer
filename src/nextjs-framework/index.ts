export * from './pages/render-page';
export * from './pages/render-widget';
export * from './pages/utils';
export * from './widgets';
export * from './rest-sdk';
export * from './services/render-widget-service';
export * from './editor';
