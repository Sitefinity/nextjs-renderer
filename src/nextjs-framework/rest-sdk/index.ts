export * from './rest-service';
export * from './root-url.service';
export * from './service-metadata';
export * from './dto';
export * from './filters';
export * from './services';
