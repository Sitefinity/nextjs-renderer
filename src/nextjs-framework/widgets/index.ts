export * from './widget-registry';
export * from './error/widget-execution-error-component';
export * from './login-form/login-form';
export * from './content-block/content-block';
export * from './classification/classification';
export * from './call-to-action/call-to-action';
export * from './breadcrumb/breadcrumb';
