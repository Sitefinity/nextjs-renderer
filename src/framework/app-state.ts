import { RequestContext } from 'sitefinity-react-framework/services/request-context';
import { WidgetModel } from 'sitefinity-react-framework/widgets/widget-model';

export interface AppState {
    requestContext: RequestContext;
    widgets: WidgetModel<any>[];
}
