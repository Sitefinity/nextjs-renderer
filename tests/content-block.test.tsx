
import { waitFor } from '@testing-library/react';
import { RenderType, WidgetTester } from './framework/widget-tester';
import { RestSdkTypes, RestService } from '../src/nextjs-framework/rest-sdk';
import { ContentBlockEntity } from '../src/nextjs-framework/widgets';

test('Content block render Content property', async () => {
    await WidgetTester.testWidgetRender<ContentBlockEntity>({
        name: 'SitefinityContentBlock',
        properties: {
            Content: 'Test content'
        },
        assert: async (element) => {
            await waitFor(() => {
                expect(element).toMatchSnapshot();
            });
        }
    });
});

test('Content block render with Content item', async () => {
    const contentItem = await RestService.createItem({
        Type: RestSdkTypes.GenericContent,
        Data: {
            Title: 'Test content block',
            Content: 'Test content block content'
        }
    });

    await WidgetTester.testWidgetRender<ContentBlockEntity>({
        name: 'SitefinityContentBlock',
        properties: {
            SharedContentID: contentItem.Id
        },
        assert: async (element) => {
            await waitFor(() => {
                expect(element).toMatchSnapshot();
            });
        }
    });
});

test('Content block html metadata assert during edit', async () => {
    await WidgetTester.testWidgetRender<ContentBlockEntity>({
        name: 'SitefinityContentBlock',
        properties: {
            Content: 'Test content'
        },
        assert: (element) => {
            const actualElement = element.querySelector('[data-sfid]');
            actualElement?.setAttribute('data-sfid', 'static');
            expect(element).toMatchSnapshot();
        },
        render: RenderType.Edit
    });
});
