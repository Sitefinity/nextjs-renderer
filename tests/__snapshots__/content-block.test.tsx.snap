// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content block html metadata assert during edit 1`] = `
<div>
  <div
    data-sfhasquickeditoperation="true"
    data-sfid="static"
    data-sfiscontentwidget="true"
    data-sfisempty="false"
    data-sfisemptyvisualhidden="false"
    data-sfisorphaned="false"
    data-sfname="SitefinityContentBlock"
    data-sftitle="Content block"
    draggable="true"
  >
    Test content
  </div>
</div>
`;

exports[`Content block render Content property 1`] = `
<div>
  <div>
    Test content
  </div>
</div>
`;

exports[`Content block render with Content item 1`] = `
<div>
  <div>
    Test content block content
  </div>
</div>
`;

exports[`Content block with custom attributes 1`] = `
<div>
  <div
    test-attribute-key="test-attribute-value"
  >
    Test content
  </div>
</div>
`;

exports[`Content block with custom tag name(CB_TagName_Rendered) 1`] = `
<div>
  <h1>
    Test content
  </h1>
</div>
`;
